app/models/user.rb

```diff
@@ -400,6 +400,14 @@ def has_devices?
     !recent_devices.empty?
   end
 
+  def new_device(cookies)
+    existing_device = cookies[:device] && Device.find_by(
+      user_id: id,
+      cookie_uuid: cookies[:device],
+    )
+    return existing_device.nil?
```

Comment:
Inside the `User` model, we could reference a user's own devices directly with the `devices` method:

```suggestion
    !cookies[:device] || !devices.exists?(cookie_uuid: cookies[:device])
```

Reply:
That's much more succinct. Implemented as suggested.